lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -714,35 +716,51 @@ def wait_for_host_in_dashboard(host)
 
         # Ensure the host has requested a cert, then sign it
         #
-        # @param [Host, Array<Host>, String, Symbol] host    One or more hosts to act upon,
+        # @param [Host, Array<Host>, String, Symbol] host   One or more hosts to validate signing certificate,
         #                            or a role (String or Symbol) that identifies one or more hosts.
+        #                            No argument, or an empty array means no validation of success
+        #                            for specific hosts will be performed.
         #
         # @return nil
         # @raise [FailTest] if process times out
-        def sign_certificate_for(host)
-          block_on host do | host |
-            if [master, dashboard, database].include? host
+        def sign_certificate_for(host = [])
```

Comment:
The name of this method is misleading.  Without looking at anything other than the method signature, I would expect that `host` is an array of hostname strings that will be passed to `puppet cert sign <hostname>`, and if I gave it an empty list then it would be a NOOP.

I see that this was already misleading prior to these changes, but since we're in there and altering the behavior, we should consider improving this.  Ideally we'd rename the method name to be more clear, but if we want to avoid "API" changes then we should at least improve the docstring a bunch.

I think the docstring should first and foremost say that it will sign all certificates, regardless of the parameter value.